sonic-gnmi-standalone/proto/firmware_management.proto

```diff
@@ -0,0 +1,296 @@
+  // If true, only return what would be removed without actually removing anything
+  bool dry_run = 1;
+}
+
+// ConsolidateImagesResponse contains the result of the consolidation operation
+message ConsolidateImagesResponse {
+  // The current image that was set as default
+  string current_image = 1;
+  
+  // List of images that were removed (or would be removed in dry_run)
+  repeated string removed_images = 2;
+  
+  // Estimated space freed in bytes (if available)
+  int64 space_freed_bytes = 3;
+  
+  // Any warnings or non-fatal errors encountered
+  repeated string warnings = 4;
+  
+  // Whether the operation was actually performed (false for dry_run)
+  bool executed = 5;
+}
+
+// ListImagesRequest is the request to list installed SONiC images
+message ListImagesRequest {
+}
+
+// ListImagesResponse contains the list of installed SONiC images
+message ListImagesResponse {
+  // List of all installed image names
+  repeated string images = 1;
+  
+  // Name of the currently running image
+  string current_image = 2;
+  
+  // Name of the next boot image
+  string next_image = 3;
+  
+  // Any warnings or non-fatal errors encountered
+  repeated string warnings = 4;
+}
+
+// Request to start firmware download
+message DownloadFirmwareRequest {
+  // URL to download from (e.g., "http://10.201.148.43/pipelines/Networking-acs-buildimage-Official/vs/test/sonic-vs.bin")
+  string url = 1;
+  
+  // Optional output path. If empty, filename is auto-detected from URL and saved to /host
+  string output_path = 2;
+  
+  // Optional connection timeout in seconds (defaults to 30)
+  int32 connect_timeout_seconds = 3;
+  
+  // Optional overall timeout in seconds (defaults to 300)
+  int32 total_timeout_seconds = 4;
+  
+  // Optional MD5 checksum for verification after download
+  string expected_md5 = 5;
+}
+
+// Response from starting firmware download
+message DownloadFirmwareResponse {
+  // Unique session ID for tracking this download
+  string session_id = 1;
+  
+  // Initial status message
+  string status = 2;
+  
+  // Resolved output path where file will be saved
+  string output_path = 3;
+}
+
+// Request to get download status
+message GetDownloadStatusRequest {
+  // Session ID returned from DownloadFirmware
+  string session_id = 1;
+}
+
+// Response with current download status
+message GetDownloadStatusResponse {
+  // Session ID
+  string session_id = 1;
+  
+  // Current download state - exactly one of these will be set
+  oneof state {
+    // Download is starting up
+    DownloadStarting starting = 2;
+    
+    // Download is in progress
+    DownloadProgress progress = 3;
+    
+    // Download completed successfully
+    DownloadResult result = 4;
+    
+    // Download failed with error
+    DownloadError error = 5;
+  }
+}
+
+// Download is starting up
+message DownloadStarting {
+  // Status message
+  string message = 1;
+  
+  // Time when download started (RFC3339 format)
+  string start_time = 2;
```

Comment:
Is there a timestamp type for Protobuf? If not, I would suggest using the Unix timestamp (`int64`) since it's representation-independent and consumes less space in the protobuf message.